templates/an-cash-dao/contracts/ANCashDAOTemplate.sol

```diff
@@ -0,0 +1,164 @@
+/*
+ * SPDX-License-Identifier:    MIT
+ */
+
+pragma solidity 0.4.24;
+
+import "./TokenCache.sol";
+import "./BaseTemplate.sol";
+
+import "./lib/os/ERC20.sol";
+
+
+contract ANCashDAOTemplate is BaseTemplate, TokenCache {
+    string constant private ERROR_MISSING_CACHE = "TEMPLATE_MISSING_TOKEN_CACHE";
+    string constant private ERROR_EMPTY_HOLDERS = "COMPANY_EMPTY_HOLDERS";
+    string constant private ERROR_BAD_HOLDERS_STAKES_LEN = "COMPANY_BAD_HOLDERS_STAKES_LEN";
+    string constant private ERROR_BAD_VOTE_SETTINGS = "COMPANY_BAD_VOTE_SETTINGS";
+    string constant private ERROR_BAD_COLLATERAL_REQUIREMENT_SETTINGS = "COMPANY_BAD_COL_REQ_SETTINGS";
+
+    bool constant private SET_APP_FEES_CASHIER = false;
+    bool constant private TOKEN_TRANSFERABLE = true;
+    uint8 constant private TOKEN_DECIMALS = uint8(18);
+    uint256 constant private TOKEN_MAX_PER_ACCOUNT = uint256(0);
+    uint64 constant private DEFAULT_FINANCE_PERIOD = uint64(30 days);
+
+    struct Cache {
+        address dao;
+        address token;
+        address tokenManager;
+        address agreement;
+    }
+
+    mapping (address => Cache) internal cache;
+
+    constructor(DAOFactory _daoFactory, ENS _ens, MiniMeTokenFactory _miniMeFactory, IFIFSResolvingRegistrar _aragonID)
+        BaseTemplate(_daoFactory, _ens, _miniMeFactory, _aragonID)
+        public
+    {
+        _ensureAragonIdIsValid(_aragonID);
+        _ensureMiniMeFactoryIsValid(_miniMeFactory);
+    }
+
+    function setupDAO(string _id, string _name, string _symbol, address[] _holders, uint256[] _stakes) external {
+        _validateId(_id);
+        (Kernel dao, ACL acl) = _createDAO();
+        MiniMeToken token = _createToken(_name, _symbol, TOKEN_DECIMALS);
+
+        TokenManager tokenManager = _installTokenManagerApp(dao, token, TOKEN_TRANSFERABLE, TOKEN_MAX_PER_ACCOUNT);
+        _mintTokens(acl, tokenManager, _holders, _stakes);
+        _storeCache(dao, token, tokenManager);
+        _registerID(_id, address(dao));
+    }
+
+    function installAgreement(string _title, bytes _content, address _arbitrator, address _stakingFactory) external {
```

Comment:
Perhaps once we simplify a bit (i.e. do this last), we can push this into `installApps()`... 🤞.

Wouldn't try very hard at this though; if it immediately gives a stack too deep, let's just keep it like this.

Reply:
It's been impossible to make it fit there, I've been trying a lot haha, will try once again tho